servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/SocketOptionUtils.java

```diff
@@ -24,13 +24,39 @@
 
 import java.net.SocketOption;
 import java.net.StandardSocketOptions;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Function;
 import javax.annotation.Nullable;
 
+import static java.util.function.Function.identity;
+
 /**
  * Utilities to convert {@link SocketOption}s.
  */
 public final class SocketOptionUtils {
+    private static final Map<SocketOption<?>, OptConverter<?>> SOCKET_OPT_MAP = new HashMap<>();
+    static {
+        putOpt(ChannelOption.IP_MULTICAST_IF, StandardSocketOptions.IP_MULTICAST_IF);
+        putOpt(ChannelOption.IP_MULTICAST_LOOP_DISABLED, StandardSocketOptions.IP_MULTICAST_LOOP,
+                SocketOptionUtils::boolNot, SocketOptionUtils::boolNot);
```

Comment:
socket to channel conversion should have a null check for the passed value to fail at the builder vs when a channel is created.

Reply:
I don't think we need to be restrictive/opinionated here. Netty treats `null` as "remove option" and if we utilize that in the future this code wouldn't need to change, and the overhead for null check is minimal vs cognitive overhead to understand why they need to be different.